src/localplayer.cpp

```diff
@@ -457,6 +450,42 @@ void LocalPlayer::move(f32 dtime, Environment *env, f32 pos_max_d,
 		setSpeed(m_speed);
 		m_can_jump = false;
 	}
+
+	// Autojump
+	if (autojump) {
+		// release autojump after a given time
+		autojump_time -= dtime;
+		if (autojump_time <= 0.0f) {
+			autojump = false;
+		}
+	} else if (m_can_jump && !control.jump && !control.sneak && control.up && g_settings->getBool("autojump")) {
```

Comment:
The autojump setting should be cached instead of reading it from settings frequently.

Reply:
Good point; do you have an example of a cached setting that I could copy? Just off the top of your head?

Reply:
I'll look. We can cache this because it is not a setting that will change in-game.

Reply:
Actually it will; it's part of the key change GUI.

Reply:
Should be solved with the `PlayerSettings` construct.